File: codeforces/835/A.py
a, b, c, d, e = map(int,input().split())

wynik_pierwszego = d + (b * a) + d
wynik_drugiego = e + (c * a) + e

if wynik_pierwszego < wynik_drugiego:
    print("First")
elif wynik_pierwszego > wynik_drugiego:
    print("Second")
elif wynik_drugiego == wynik_pierwszego:
    print("Friendship")

File: codeforces/835/A.cpp
#include<bits/stdc++.h>
#include<iostream>
#include<cstring>
#include<iomanip>
#include<fstream>

using namespace std;

int main()

{
    int a,b,c,d,e;
    cin >> a >> b >> c >> d >> e;
    int wynik_jeden = d + (b * a) + d;
    int wynik_dwa = e + (c * a) + e;
    if(wynik_jeden > wynik_dwa)
        cout << "Second";
    if(wynik_dwa > wynik_jeden)
        cout << "First";
    if(wynik_dwa == wynik_jeden)
        cout << "Friendship";


}